3D margins consistency  for mplot3d (isometric projection)

neok-m4700 opened the pull request:
**mwe**
```python
from mpl_toolkits.mplot3d import proj3d
import pylab as plt
import numpy as np
import itertools
import math

# the following code snippet should plot a 3D cube, with apparent equal edge lengths
kw_3d = dict(proj_type='ortho', projection='3d', aspect='equal')

_, ax = plt.subplots(subplot_kw=kw_3d)
ax.grid(1)

# draw cube (stackoverflow.com/a/11156353)
r = (-1, 1)
for s, e in itertools.combinations(np.array(list(itertools.product(r, r, r))), 2):
    if abs(s - e).sum() == r[1] - r[0]:
        ax.plot3D(*zip(s, e), c='C0')

# isometric projection (en.wikipedia.org/wiki/Isometric_projection)
ax.view_init(elev=math.degrees(math.atan(1 / math.sqrt(2))), azim=-45)

# annotation
px, py, pz = 1, -1, 1
ax.plot([px], [py], [pz], 'o', ms=10, mfc='none', c='C1')

x, y, _ = proj3d.proj_transform(px, py, pz, ax.get_proj())
xt, yt, _ = proj3d.proj_transform(px, py + .6, pz + .6, ax.get_proj())

ax.annotate('Isometric', (x, y), xytext=(xt, yt), arrowprops=dict(arrowstyle='->'))

plt.savefig('cube.png', dpi=300)
```

**without proposed patch**
![cube_nok](https://user-images.githubusercontent.com/13423344/52221436-776d2c80-28a1-11e9-979f-e7eeb874bed4.png)

**with current PR**
![cube_ok](https://user-images.githubusercontent.com/13423344/52221458-805dfe00-28a1-11e9-86bc-f7edfb8f3c1b.png)

<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.

- Help with git and github is available at
  https://matplotlib.org/devel/gitwash/development_workflow.html.

- Do not create the PR out of master, but out of a separate branch.

- The PR title should summarize the changes, for example "Raise ValueError on
  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
  "Addresses issue #8576".

- The summary should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues.

- If you are contributing fixes to docstrings, please pay attention to
  http://matplotlib.org/devel/documenting_mpl.html#formatting.  In particular,
  note the difference between using single backquotes, double backquotes, and
  asterisks in the markup.

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

timhoffm commented on lib/mpl_toolkits/mplot3d/axes3d.py:
```diff
@@ -82,6 +82,7 @@ def __init__(
 
         self.initial_azim = azim
         self.initial_elev = elev
+        self.proj_type = proj_type
```
Why this? it gets overwritten by `set_proj_type` in the next line. If you just want to define `self.proj_type` in `__init__`, I'd rather set it to `None`.

QuLogic commented:
This seems to have been forgotten for a while, but I don't see why it shouldn't go in. It will need a rebase though.

I also wonder what effect there would be if the margin were always added (for defaults), even for perspective projections.

neok-m4700 commented:
> This seems to have been forgotten for a while, but I don't see why it shouldn't go in. It will need a rebase though.

Hi @QuLogic and @timhoffm, I squashed all the previous commits and updated this branch against latest master.
All tests are passing so I think that this (old) PR is ready to merge, if you don't see any objection.
